R/multitables.R

```diff
@@ -26,6 +26,60 @@ setMethod("[[", c("MultitableCatalog", "numeric"), function (x, i, ...) {
     getEntity(x, i, Multitable, ...)
 })
 
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "character", "missing", "formula"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              
+              ds <- loadDataset(datasetReference(x))
+              template <- formulaToQuery(value, data = ds)
+              
+              if (i %in% names(x)) {
+                  payload <- makeMultitablePayload(template = template)
+                  crPATCH(urls(x)[match(i, names(x))], body=toJSON(payload))
+                  return(x)
+              } else {
+                  ## Creating a new filter
+                  f <- .newMultitable(template, self(x), name = i)
+                  return(refresh(x))
+              }
+          })
+
+makeMultitablePayload <- function (template, name) {
+    template <- lapply(template$dimensions,
+                       function (x) list(query=x))
+
+    if (missing(name)) {
+        return(wrapEntity(template = template))
+    } else {
+        return(wrapEntity(name = name, template = template))
+    }
+}
+
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "character", "missing", "NULL"),
+          function (x, i, j, value) {
+              stopifnot(length(i) == 1)
+              if (i %in% names(x)) {
+                  if (is.null(value)) {
+                      delete(x[[i]])
+                      return(invisible(NULL))
+                  }}
+          })
+
+#' @rdname catalog-extract
+#' @export
+setMethod("[[<-", c("MultitableCatalog", "numeric", "missing", "NULL"),
```

Comment:
With`c("MultitableCatalog", "ANY", "missing", "NULL")` we don't get an error, but line 60 needed to be modified to be true whenever there's a numeric.